lib/src/configuration.dart

```diff
@@ -36,14 +36,23 @@ class Configuration {
   RealmSchema get schema => _schema;
 
   /// Creates a [Configuration] with schema objects for opening a [Realm].
+  ///
   /// [readOnly] controls whether a [Realm] is opened as readonly.
   /// This allows opening it from locked locations such as resources,
   /// bundled with an application.  The realm file must already exists.
-  Configuration(List<SchemaObject> schemaObjects, {bool readOnly = false})
+  ///
+  /// [fifoFilesFallbackPath] is a custom directory for storing FIFO special files
+  /// in case the Realm file is in a location that does not allow the creation of FIFO special files.
+  Configuration(List<SchemaObject> schemaObjects, {String? fifoFilesFallbackPath, bool readOnly = false})
       : _schema = RealmSchema(schemaObjects),
         _handle = realmCore.createConfig() {
     schemaVersion = 0;
     path = defaultPath;
+
+    if (fifoFilesFallbackPath?.isNotEmpty ?? false) {
+      this.fifoFilesFallbackPath = fifoFilesFallbackPath!;
+    }
+
     if (readOnly) {
```

Comment:
Not part of this PR, but I noticed this check. Should we not just set property regardless?

Reply:
FIFO path is empty by default in C-API. It doesn't need to be set. This is some kind as a small optimization.
If you mean ReadOnly, it is a mode and the value is enum value. I would like to keep the default enum value of schema mode, which is currently "Automatic". But you are right it is not visible here, you can see it inside _realmLib.realm_config_set_in_memory.

Reply:
Okay. Should we have an enum here as well then? (eventually)

Reply:
there is sense in doing that. The least we call core the better. I think this is should be the case for all optional properties.